### DISCUSSION TITLE
How to make circle border like this in Png

### DISCUSSION THREAD

**Discussion by AmjadGhzlangit** [chillerlan/php-qrcode#267] [Q&A] [answered]
![Uploading photo_2024-05-24_00-57-49.jpg…]()

**Comment by codemasher** [comment]
I don't see the image you uploaded but I assume it's a screenshot of the round quietzone SVG example?

Either way, the simple answer is: don't. Render your QR Code in SVG and convert it as shown in several examples.
The more complicated answer: draw a circle in GD or Imagick using similar math as in the SVG example.

**Reply by AmjadGhzlangit** [reply]
sorry this is the image
![photo_2024-05-24_00-57-49](https://github.com/chillerlan/php-qrcode/assets/96890984/9cd5092a-c661-47d8-9e54-bece7c4a2ff2)

**Reply by AmjadGhzlangit** [reply]
@codemasher

how to customize the shape of FINDER? please

**Reply by codemasher** [reply, marked as answer]
You don't need to double ping me, time zones exist.

There is an SVG example for exactly this over here https://github.com/chillerlan/php-qrcode/blob/v5.0.x/examples/svgWithLogoAndCustomShapes.php

It would work a bit different and probably more complex in GD/Imagemagick: you would need to skip the generation of the finder patterns in the respective `module()` methods, and then draw replacements (or insert images) in their positions. 

You can calculate the positions from the image size:

- the width/height of the finder patterns is `7 * $this->scale`
- the width of the QR symbol (excluding quietzone) is `$this->matrix->getVersion()->getDimension() * $this->scale` 
- the quietzone width is `$this->options->quietzoneSize * $this->scale`

**Reply by AmjadGhzlangit** [reply]
thanks
